server/ot_rpc.go

```diff
@@ -73,23 +77,40 @@ func runTranscoder(n *core.LivepeerNode, orchAddr string, capacity int) error {
 		}
 		return err
 	}
+	exitc := make(chan os.Signal)
+	signal.Notify(exitc, os.Interrupt)
+	defer signal.Stop(exitc)
+	go func() {
+		select {
+		case sig := <-exitc:
+			glog.Infof("Exiting Livepeer Transcoder: %v", sig)
+			// Cancelling context will close connection to orchestrator
+			cancel()
+			return
+		}
+	}()
 
 	httpc := &http.Client{Transport: &http.Transport{TLSClientConfig: &tls.Config{InsecureSkipVerify: true}}}
+	var wg sync.WaitGroup
 	for {
 		notify, err := r.Recv()
 		if err != nil {
 			status := status.Convert(err)
 			if status.Message() == SecretErr.Error() { // consider this unrecoverable
 				return SecretErr
 			}
+			glog.Infof("End of stream recieve cylcle because err=%v, waiting for running transcode jobs to complete", err)
+			wg.Wait()
 			return err
 		}
-		go runTranscode(n, orchAddr, httpc, notify)
+		wg.Add(1)
+		go runTranscode(n, orchAddr, httpc, notify, &wg)
```

Comment:
Could probably do this to keep `wg` local and avoid the need to propagate it:

```
wg.Add(1)
go func() {
  runTranscode(n, orchAddr, httpc, notify)
  wg.Done()
}()
```

Reply:
Fixed